### ISSUE TITLE
Add favicon support

### ISSUE THREAD

**Issue by Martinsos** [wasp-lang/wasp#9]
We want Wasp to have support for adding a favicon. This should be property of `app` declaration:
```
app MyApp {
  title: "My super app!",
  favicon: "@ext/favicon.ico"
```
Should favicon really go into `ext/` or not (and go somewhere next to .wasp), is debatable. Also, what are the requirements on favicon? Does it have to have certain dimensions? What about specifying different favicon of different qualities, as can be normal for some cases? What about some additional properties?

It feels like this should be researched more -> understanding the essence of favicon, and then we can go from there.

**Comment by Martinsos** [comment]
@matijaSos added support for custom code in the `<head>` -> maybe this takes care of favicon, to some degree?

**Comment by shayneczyzewski** [comment]
Ok, so this turned out to be slightly more complex than originally described. :D A few notes on the complications and next steps based on a quick chat with @matijaSos and @sodic:
- We can't really just list a single file, as there can be _many_ different favicon files for different sizes and targets. And while they are often named sanely, it may be limiting to enforce a specific pattern for them.
  - Exe: https://evilmartians.com/chronicles/how-to-favicon-in-2021-six-files-that-fit-most-needs
  - So we could turn that from a single path into a list of paths, but that becomes cumbersome fast.
- Users also need to be able to update the `manifest.json` file as well. So is that another field? Or inferred? Etc.
- The idea I proposed instead is to add support for a `public/` dir, and we copy/merge the contents of that user dir with our defaults (including using our `index.html` file). In this way, they can add many different favicons with any names, customize the manifest, and add any other static images (right now we only support relative imports in React components, which is not ideal).
- As for where this new directory, and other things like `tailwind.config.js`/`Dockerfile` should live- we may want to review our project structure again. I will create a new issue for this task, but the short overview is: should we go from `src/client` etc. to `client/src` so then we could put configs under the appropriate area, and then keep `public` at that top level?
  - Aside: the answer to this impacts how we copy over `ExternalCode.File`s.

TL;DR All of this requires a bit more thought and design (and time to implement), so it will be left for after Beta so we don't rush a bad idea. The only pre-Beta task will be changing the default stick to our actual logo. I will make a PR for that shortly.
